src/screens/Inventory/AddInventory.tsx

```diff
@@ -0,0 +1,111 @@
+import { withStyles } from '@material-ui/core/styles';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import BaseScreen from '../../components/BaseComponents/BaseScreen';
+import Button from '../../components/Button';
+import TextField from '../../components/TextField';
+import { InventoryRecord, ProductRecord, SiteRecord } from '../../lib/airtable/interface';
+import { createInventory } from '../../lib/airtable/request';
+import { addInventoryToRedux } from '../../lib/redux/inventoryData';
+import { EMPTY_INVENTORY, productIdString, siteIdString, SiteInventoryData } from '../../lib/redux/inventoryDataSlice';
+import { EMPTY_SITE } from '../../lib/redux/siteDataSlice';
+import { RootState } from '../../lib/redux/store';
+import { generateOfflineInventoryId } from '../../lib/utils/inventoryUtils';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    content: {
+      color: theme.palette.text.primary,
+    },
+    formControl: {
+      width: '100%',
+    },
+  });
+
+interface AddInventoryProps extends RouteComponentProps {
+  classes: { content: string, formControl: string};
+  location: any;
+  currentSite: SiteRecord;
+  products: Record<productIdString, ProductRecord>;
+  userId: string;
+  sitesInventory: Record<siteIdString, SiteInventoryData>;
+}
+
+
+function AddInventory(props: AddInventoryProps) {
+  const { classes, currentSite, products, userId, sitesInventory } = props;
+  const history = useHistory();
+  // Products that the site already has inventory for
+  const currentSiteProducts = sitesInventory[currentSite.id].siteInventory.map((inventory: InventoryRecord) => inventory.productId);
+  {/* Only show products that the site doesn't already have */}
+  const productOptions = Object.entries(products).filter(([id, _]) => !currentSiteProducts.includes(id)).map(item => item[0]);
+  const [selectedProductId, setSelectedProductId] = useState("");
+  const [startingAmount, setStartingAmount] = useState(0);
+
+  // TODO: Add form input validation and error messaging
+  const handleSubmit = async (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    // Make a deep copy of an empty inventory record
+    const inventory = JSON.parse(JSON.stringify(EMPTY_INVENTORY));
+    inventory.productId = selectedProductId;
+    inventory.siteId = currentSite.id;
+    inventory.currentQuantity = startingAmount;
+    inventory.periodStartQuantity = startingAmount;
+    
+    let inventoryId = "";
+    // Grab new inventory id from API response or generate temporary ID if offline
+    try {
+      await createInventory(inventory).then(resp => resp?.json()).then(data => inventoryId = data.id);
+    } catch (err) {
+      inventoryId = generateOfflineInventoryId();
+      console.error("[AddInventory] (handleSubmit) ", err);
+    }
```

Comment:
This is my fault (I set a bad precedent with `addCustomer` etc..) but I don't think this try catch does what you want it to do (though it does work). 

Here's the error:
![image](https://user-images.githubusercontent.com/35501399/112037864-6de5c380-8aff-11eb-8932-9701cf935f66.png)

Since the error is already caught in the function in `request.js`, the function `createInventory` just returns null and we error again doing `data.id`. We could remove error handling on this level and put it all in `request.js` level (also moving `addInventoryToRedux` to that level as well probably like mentioned in [comments](https://github.com/calblueprint/meepanyar/pull/55/files?file-filters%5B%5D=.tsx#r598162072) in tiffany's PR) OR we could remove it on the request level and do it on this level. I probably prefer the former.

Either way, we could also save this for another PR and do it all at once later in the week (since this needs to be done for `addCustomer` as well

Reply:
Went with the former and moved addInventoryToRedux into `createInventory`